src/vsphere_cpi/lib/cloud/vsphere/storage_picker.rb

```diff
@@ -38,5 +38,49 @@ def choose_persistent_storage(size_in_mb, target_datastore_pattern, available_da
       datastore_name = datastore_picker.pick_datastore_for_single_disk(disk_config)
       datastore_name
     end
+
+    # choose storage from set of datastore and sdrs enabled datastore clusters
+    # using weight based algorithm on free space to choose the storage
+    # if datastore clusters are provided and none of them have sdrs enabled log an error
+
+    # @param [String] target_datastore_name
+    # @param [Hash] accessible_datastores
+    # @param [VmType] vm_type
+    def choose_ephemeral_storage(target_datastore_name, accessible_datastores, vm_type, logger)
+      #pick datastore set by cluster_picker based on ephemeral_pattern
+      storage_options = [accessible_datastores[target_datastore_name]].compact
```

Comment:
This looks off. Should it be `Array(accessible_datastores[target_datastore_name])` instead?

Reply:
This one lets chat tomorrow, as I was also undecided if I should directly pass the actual datastore here instead of first 2 parameters